src/PIL/JpegImagePlugin.py

```diff
@@ -138,8 +138,11 @@ def APP(self, marker):
     elif marker == 0xFFEE and s[:5] == b"Adobe":
         self.info["adobe"] = i16(s, 5)
         # extract Adobe custom properties
+        # https://exiftool.org/TagNames/JPEG.html#Adobe
+        # https://docs.oracle.com/javase/8/docs/api/javax/imageio/metadata/doc-files/jpeg_metadata.html
         try:
-            adobe_transform = s[1]
+            index2_offset = 5
+            adobe_transform = s[index2_offset + 3]  # Index2 3
```

Comment:
This is actually incorrect as well. As per https://github.com/LuaDist/libjpeg/blob/6c0fcb8ddee365e7abc4d332662b06900612e923/jcmarker.c#L435-L443, the layout is
- 5 bytes for "Adobe" (as evidence a few lines earlier in this Pillow file in `s[:5] == b"Adobe":`
- 2 bytes for DCTEncodeVersion
- 2 bytes for APP14Flags0
- 2 bytes for APP14Flags1
- 1 byte for transform

This makes `s[11]` transform. 

I've created #5408 for this.